reader.go

```diff
@@ -79,6 +80,29 @@ func NewDialectReader(r io.Reader, opts Dialect) *Reader {
 	}
 }
 
+// create new reader that detects BOM and removes it as necessary
```

Comment:
Remove commented out code between 83 and 105.
Is there a reason that you want to keep the old code around?

Reply:
No reason to keep it. Fixed this.